api/langchain_util.py

```
@@ -164,4 +164,192 @@ async def get_rag_chain(query: str, chat_history: List[Dict], tools: List[Any] =
+        
+        question_answer_chain = create_stuff_documents_chain(llm, qa_prompt)
+        runnable = create_retrieval_chain(history_aware_retriever, question_answer_chain)
+
+    # Catch-all
+    if not runnable:
+         yield {"type": "error", "error": "Failed to initialize chain"}
+         return
+
+    # Execute with Streaming
+    final_output = ""
+    accumulated_tools = []
```

Ruff (0.14.14)

[error] Local variable `accumulated_tools` is assigned to but never used

Remove assignment to unused variable `accumulated_tools`

(F841)